### PR TITLE
HOTT-3747 - replacing the old redis v5.x cluster with new multi-node redis v7.x cluster

### PR DESCRIPTION
**Pull request by venkatp227**
# Pull Request

## What?

I have:

- aws provider version upgraded to >= 5.3
- Added new redis cluster terraform module
- Replaced old single node redis v5.x cluster with new multi-node redis v7.x cluster

## Why?

- Old redis cluster connectivity issues

### CODE REVIEW

File: environments/development/common/redis.tf [hunk 1]
```diff
@@ -6,26 +6,103 @@ locals {
...
+}
+
+# tfsec:ignore:aws-cloudwatch-log-group-customer-key
+resource "aws_cloudwatch_log_group" "redis_engine_lg" {
+  for_each          = local.redis
+  name              = "redis-${each.key}-${var.environment}-engine-lg"
+  retention_in_days = 30
+}
+
+### Single node redis cluster - v1 deployment (resulted in connectivity issues due to endpoints being blank)
```
Comment (willfish) [inline environments/development/common/redis.tf:28]: I guess we can just remove this?

File: environments/development/common/redis.tf [hunk 2]
```diff
@@ -6,26 +6,103 @@ locals {
...
+  node_type                   = "cache.t3.micro"
+  security_group_ids          = [module.alb-security-group.redis_security_group_id]
+  subnet_group_name           = aws_elasticache_subnet_group.this.name
+  multi_az_enabled            = true
+  automatic_failover_enabled  = true
+  preferred_cache_cluster_azs = ["eu-west-2a", "eu-west-2b", "eu-west-2c"]
 
-  security_group_ids = [module.alb-security-group.redis_security_group_id]
-  subnet_group_name  = aws_elasticache_subnet_group.this.name
+  ## To take up enabling redis encryption after migration
```
Comment (willfish) [inline environments/development/common/redis.tf:80]: :+1:

Comment (willfish) [reply environments/development/common/redis.tf:80]:
```suggestion
  # To take up enabling redis encryption after migration
```

**Review by willfish — APPROVED** [review]
Tiny comments but I'm excited for us to be using redis v7!

File: environments/development/common/redis.tf (hunk 1 above)
Comment (venkatp227) [reply environments/development/common/redis.tf:28]: I hope this comment is about removing the `### Single node redis cluster - v1 deployment` and not the retention_in_days @willfish

Comment (willfish) [reply environments/development/common/redis.tf:28]: Yeah, just the comment :smile:

File: environments/development/common/redis.tf (hunk 2 above)
Comment (venkatp227) [reply environments/development/common/redis.tf:80]: updated

**Pull request merged**